src/library/autodj/autodjprocessor.cpp

```diff
@@ -586,6 +587,20 @@ bool AutoDJProcessor::removeTrackFromTopOfQueue(TrackPointer pTrack) {
     if (m_pConfig->getValueString(ConfigKey(kConfigKey, "Requeue")).toInt()) {
         m_pAutoDJTableModel->appendTrack(nextId);
     }
+
+    // Fill random tracks if configured
+    int minAutoDJCrateTracks = m_pConfig->getValueString(
+            ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+    bool randomQueueEnabled = (((m_pConfig->getValueString(
+            ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+
+    bool minTracksLeft = (m_pAutoDJTableModel->rowCount() <= minAutoDJCrateTracks);
+
+    if (randomQueueEnabled && minTracksLeft) {
+        qDebug() << "Randomly adding tracks";
+        emit(randomTrackRequested(minTracksLeft));
```

Comment:
Maybe I missread you code, so please correct me if I am wrong: 
bool minTracksLeft could be something like bool trackCountBelowMinimum. 
The signal is only send when this is true, so you need not transmit that bool by the signal. 

You may change the logic to: 
int tracksToAdd = minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount();
if (randomQueueEnabled && tracksToAdd > 0) {